### PR TITLE
Ticket 4123 standard court orders btn

### LINKED ISSUE #4123: Add a button to add a Standard Court Order to a Casa Case
**What type(s) of user does this feature affect?**
 - volunteers
 - supervisors
 - admins

**Description**

- [ ] On the Edit Casa Case page  like https://casa-qa.herokuapp.com/casa_cases/CINA-14-1002/edit add a second button like the "Add a court order" button which says "Add a standard court order" 
- [ ] When the + button with the "Add a standard court order"  text is clicked, show a modal or dropdown (try em and see what feels better) of options. For now make the list hardcoded in the code- see list below
- [ ] When a standard court order is clicked, add a court order with that text and stay on the edit casa case page. 
- [ ] (Optional) make that section look a bit nicer: make court orders a proper title, make the buttons more even, anything else that might bother you

Ex:

image

(In a later ticket we will add a db table for customization and the db seed to auto setup standard court orders)

Standard court orders:

- Individual therapy for the Respondent;
- Family therapy;
- Birth certificate for the Respondent’s;
- Educational or Vocational referrals;
- Independent living skills classes or workshops;
- Learners’ permit for the Respondent, drivers’ education and driving hours when needed;
- Educational monitoring for the Respondent,
- Tutor for the Respondent;
- Individual therapy for the [parent];
- Substance abuse assessment for the [parent];
- Housing support for the [parent];
- Visitation assistance for the Respondent to see [family];
- No contact with (mother, father, other guardian)
- Supervised visits
- Supervised visits at DSS
- Virtual Visits
- Therapy (child, mother, father, other guardian)
- Psychiatric Evaluation and follow all recommendations  (child, mother, father, other guardian)
- Substance Abuse Evaluation and follow all recommendations (child, mother, father, other guardian)
- Substance Abuse Treatment (child, mother, father, other guardian)
- Urinalysis (child, mother, father, other guardian)
- Parenting Classes (mother, father, other guardian)
- Domestic Violence Education/Group

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by dalmaboros**
### What github issue is this PR for, if any?
Resolves #4123 

### What changed, and _why_?
Initially we followed the ticket suggestion to create a separate "Add a standard court order" button, but the UX was poopoo, so we implemented a single button ("Add a court order") combined with a dropdown of options for standard court orders, as well as an option to "Create a custom court order." Choosing any standard court order option appends a court order input field above, populated with the text of the standard court order. Choosing the custom court order appends a blank court order input field above. This feature makes it a bit easier for volunteers to set up court orders.

### How is this **tested**? (please write tests!) 💖💪
- System specs in `spec/system/casa_cases/edit_spec.rb`, `spec/system/casa_cases/edit_spec.rb`
- One new class method spec in `spec/models/case_court_order_spec.rb`

### Screenshots please :)
Edit case details page:
Screenshot 2024-06-02 at 7 40 01 AM
Edit court date page:
Screenshot 2024-06-02 at 7 40 54 AM

### Feelings gif (optional)
happy turtle

### CODE REVIEW

**Review by schoork — CHANGES_REQUESTED** [review]

File: app/javascript/controllers/extended_nested_form_controller.js [hunk 1]
```diff
@@ -12,6 +14,16 @@ export default class extends NestedForm {
     }
   }
 
+  addCourtOrder (e) {
```
[the remark this answers is not in the document]
Comment (elasticspoon) [reply app/javascript/controllers/extended_nested_form_controller.js:17]: `addCourtOrder` could be `add`.

But I'm a bit confused by this controller. `extended-nested-form` implies to be that the behavior should be pretty generic but all the extensions made to the stimulus nest form controller seem very court report focused.

I initially thought based on the name it was being using in more places than with just court orders. I feel like either it should have a generic name `extended-nested-controller` and specific methods specific to court orders removeCourtOrder or have a more specific name `court-order-controller` and more generic method names.

[the remark this answers is not in the document]
Comment (compwron) [reply app/javascript/controllers/extended_nested_form_controller.js:17]: I like the longer name "addCourtOrder" tbh, for searchability - but I haven't looked deeply

File: app/models/case_court_order.rb [hunk 2]
```diff
@@ -1,5 +1,30 @@
 class CaseCourtOrder < ApplicationRecord
   IMPLEMENTATION_STATUSES = {unimplemented: 1, partially_implemented: 2, implemented: 3}
+  STANDARD_COURT_ORDERS = [
+    "Individual therapy for the Respondent",
```
Comment (compwron) [inline app/models/case_court_order.rb:4]: Please alphabetize?

**Review by schoork — APPROVED** [review]
Yeah, I do like the changes to this controller, including the name. Good work @dalmaboros!

**Comment by elasticspoon** [comment]
:partying_face:

**Pull request merged**